src/ol_openedx_checkout_external/views.py

```diff
@@ -0,0 +1,96 @@
+"""Views for External Checkout"""
+
+import logging
+
+from common.djangoapps.course_modes.models import CourseMode
+from django.conf import settings
+from django.http import HttpResponseRedirect
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import (
+    SessionAuthenticationAllowInactiveUser,
+)
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+
+log = logging.getLogger(__name__)
+
+
+class ExternalCheckoutView(DeveloperErrorViewMixin, APIView):
```

Comment:
I don't think this should be an APIView, because if something goes wrong the user is going to just get JSON back in their browser rather than a html styled error page. I think this changes some of the response types later on, which I'll make note of what I'm thinking those should be.